dlt/common/runtime/segment.py

```diff
@@ -23,24 +23,30 @@
 _SESSION: requests.Session = None
 _WRITE_KEY: str = None
 _SEGMENT_REQUEST_TIMEOUT = (1.0, 1.0)  # short connect & send timeouts
-_SEGMENT_ENDPOINT = "https://api.segment.io/v1/track"
+_SEGMENT_ENDPOINT: str = None
 _SEGMENT_CONTEXT: TExecutionContext = None
 
 
 def init_segment(config: RunConfiguration) -> None:
-    assert (
-        config.dlthub_telemetry_segment_write_key
-    ), "dlthub_telemetry_segment_write_key not present in RunConfiguration"
+    global _WRITE_KEY, _SESSION, _SEGMENT_ENDPOINT
+    if "RUNTIME__TELEMETRY_ENDPOINT" not in os.environ:
+        assert (
+            config.dlthub_telemetry_segment_write_key
+        ), "dlthub_telemetry_segment_write_key not present in RunConfiguration"
+        # store write key
+        key_bytes = (config.dlthub_telemetry_segment_write_key + ":").encode("ascii")
+        _WRITE_KEY = base64.b64encode(key_bytes).decode("utf-8")
+        # store endpoint
+        _SEGMENT_ENDPOINT = config.dlthub_telemetry_endpoint
+    else:
+        _SEGMENT_ENDPOINT = os.environ["RUNTIME__TELEMETRY_ENDPOINT"]
+        _WRITE_KEY = None
```

Comment:
if the checks mentioned above passed, you can just set _SEGMENT_ENDPOINT and _WRITE_KEY to the values form the config at the end